docs/DOCUMENTATION.md

```diff
@@ -0,0 +1,976 @@
+
+Returns: CapDataString by calling vstorage.readLatest().
+
+<br>
+<div id='middleware'></div>
+
+### <u>oracle/middleware.js</u>
+
+This file contains all the functions which are used both by the middleware. 
+
+<br>
+<div id='envvarsmiddleware'></div>
+
+<b>Environment Variables</b>
+
+This script makes use of the following environment variables and it requires them in order to function. In fact, it contains validation upon entry to make sure they are well defined.
+
+| Variable Name        	| Description                                                                                                                                                          	| Default value              	|
+|----------------------	|----------------------------------------------------------------------------------------------------------------------------------------------------------------------	|----------------------------	|
+| PORT                 	| The port on which the middleware will listen <br>for job updates or results from the CL node                                                                         	| 3000                       	|
+| EI_CHAINLINKURL      	| The CL node URL in order to connect to its API<br>to listen for jobs and send job requests.<br><b>Note that this has no default value and needs<br>to be defined</b> 	| N/A                        	|
+| FROM                 	| The address of the oracle from which to push prices.<br><b>Note that this has no default value and needs<br>to be defined</b>                                        	| N/A                        	|
+| SUBMIT_RETRIES       	| The number of retries to try when submitting a price<br>on-chain and it fails                                                                                        	| 3                          	|
+| BLOCK_INTERVAL       	| The block time of the chain in seconds. This is used<br>to query the price and round at every interval.                                                              	| 6                          	|
+| SEND_CHECK_INTERVAL  	| The interval in seconds which is waited between each send.                                                                                                           	| 45                         	|
+| AGORIC_RPC           	| The Agoric's node RPC endpoint                                                                                                                                       	| http://0.0.0.0:26657       	|
+| STATE_FILE           	| The path to the middleware state's file                                                                                                                              	| data/middleware_state.json 	|
+| CREDENTIALS_FILE     	| The path to the file containing the credentials to the <br>CL node                                                                                                   	| config/ei_credentials.json 	|
```

Comment:
You also assigning FEEDS_FILE = "oracle/feeds.json" as environment variable in middleware.js file but it is not listed here

Reply:
Removed this from an env var